### PR TITLE
[NJT] Inline through torch.nested.nested_tensor_from_jagged instead of graph break

### PR DESCRIPTION
**Pull request by soulitzer**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #124343

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang

### CODE REVIEW

**Review by jbschlosser — APPROVED** [review]
lgtm thanks!

**Comment by DanilBaibak** [comment]
Hi @soulitzer! Sorry, I need to revert your PR because it broke trunk. Here you can find more information about the issue - 
https://hud.pytorch.org/pytorch/pytorch/commit/ef93402f619f58d651845981ccd1eba1d68da077

File: test/profiler/test_profiler.py [hunk 1]
```diff
@@ -664,6 +664,9 @@ def fn(nt):
             for i in range(3):
                 values = torch.rand((8 + i, 4 + i))
                 offsets = torch.tensor([0, 2, 4, 6, 8 + i])
+                # Need to explicitly graph break due to:
+                # https://github.com/pytorch/pytorch/issues/124500
+                torch._dynamo.graph_break()
```
Comment (davidberard98) [inline test/profiler/test_profiler.py:669]: instead can we just `@skipIfTorchDynamo`? I don't think we really need to support torch.compile-ing a function that includes a profiler/execution trace

**Pull request closed without merging**